datadog_cluster_agent/datadog_checks/datadog_cluster_agent/check.py

```diff
@@ -0,0 +1,47 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from datadog_checks.base import OpenMetricsBaseCheck
+
+
+class DatadogClusterAgentCheck(OpenMetricsBaseCheck):
+    """
+    Collect Cluster Agent metrics from its Prometheus endpoint
+    """
+
+    def __init__(self, name, init_config, instances):
+        metrics = {
+            'admission_webhooks_certificate_expiry': 'admission_webhooks_certificate_expiry',
+            'admission_webhooks_mutation_attempts': 'admission_webhooks_mutation_attempts',
+            'admission_webhooks_reconcile_success': 'admission_webhooks_reconcile_success',
+            'admission_webhooks_webhooks_received': 'admission_webhooks_webhooks_received',
+            'cluster_checks_busyness': 'cluster_checks_busyness',
+            'cluster_checks_configs_dangling': 'cluster_checks_configs_dangling',
+            'cluster_checks_configs_dispatched': 'cluster_checks_configs_dispatched',
+            'cluster_checks_nodes_reporting': 'cluster_checks_nodes_reporting',
+            'cluster_checks_rebalancing_decisions': 'cluster_checks_rebalancing_decisions',
+            'cluster_checks_rebalancing_duration_seconds': 'cluster_checks_rebalancing_duration_seconds',
+            'cluster_checks_successful_rebalancing_moves': 'cluster_checks_successful_rebalancing_moves',
+            'cluster_checks_updating_stats_duration_seconds': 'cluster_checks_updating_stats_duration_seconds',
+            'external_metrics_delay_seconds': 'external_metrics_delay_seconds',
+            'external_metrics_processed_value': 'external_metrics_processed_value',
+            'rate_limit_queries_limit': 'rate_limit_queries_limit',
+            'rate_limit_queries_period': 'rate_limit_queries_period',
+            'rate_limit_queries_remaining': 'rate_limit_queries_remaining',
+            'rate_limit_queries_reset': 'rate_limit_queries_reset',
+        }
+        default_namespace = 'datadog.cluster_agent'
+        default_instances = {
+            'datadog.cluster_agent': {
+                'namespace': default_namespace,
+                'metrics': [metrics],
+                'send_monotonic_counter': False,
```

Comment:
Any reason not to submit Prometheus counters as monotonic count?

Reply:
`ddev validate metadata` does not seem to like it:

```
$ ddev validate metadata datadog_cluster_agent 
datadog_cluster_agent:10 `monotonic_count` is an invalid metric_type.
datadog_cluster_agent:12 `monotonic_count` is an invalid metric_type.
```

I also tried keeping it as a gauge in the metrics, but then `assert_metrics_using_metadata` in the tests fails because it expects a gauge, but gets a monotonic count.

Reply:
Ahh the metadata.csv type for those metrics should be `count`, let me know if that works for you. Having the counter types submitted as a gauge has been an issue before as the behavior is technically incorrect.